p2p/src/main/java/bisq/network/p2p/peers/getdata/RequestDataHandler.java

```diff
@@ -240,7 +242,12 @@ public void onMessage(NetworkEnvelope networkEnvelope, Connection connection) {
                                     // Processing 82645 items took now 61 ms compared to earlier version where it took ages (> 2min).
                                     // Usually we only get about a few hundred or max. a few 1000 items. 82645 is all
                                     // trade stats stats and all account age witness data.
-                                    dataStorage.addPersistableNetworkPayloadFromInitialRequest(e);
+
+                                    // We only apply it once from first response
+                                    if (!initialRequestApplied) {
```

Comment:
Why do we only handle the first requests for the PersistableNetworkPayload objects, but all of them for the ProtectedStoragePayload objects? Do we expect more inconsistencies between the ProtectedStorageEntrys so we use all of them? Or, were the PersistableNetworkPayload handlers not dealing with duplicates well?

I haven't been able to follow the entire TradeStatstics issue, but I am just trying to understand more about what is expected/unexpected in this startup path.

Reply:
The hash issue was caused by the extraMap where at certain updates data got added and combined with the republishing of the objects created dupicates entries as the hashes have been different due the added entries in that map (was used for tracking usage of arbitrators and mediators for a optimized selection of them). 

The optimisation for handling the PersistableNetworkPayload objects was to avoid performance spikes when processing a large set of PersistableNetworkPayload objects. For ProtectedStorageEntrys we do not expect such high amount of objects.  PersistableNetworkPayload are mainly Tradestatistic objects and AccountAgeWitnes objects and if a user have not been online for longer time those can be several 1000 objects which caused longer processing time if the normal method was used where unnecessary checks (as we got those obejcts at startup and not at runtime later) have caused most of the cpu costs.

Reply:
Ok, I think I understand. So we can have up to two loops through the ProtectedStoragePayload path and only one through the PersistableNetworkPayload path. The code handles duplicates just fine, but processing 2x the PersistableNetworkPayload objects takes a lot of time and we don't expect processing them from the second seed to provide meaningful data. Thanks for the rundown.